src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -234,66 +235,24 @@ public void initializeFromItineraryAndQueryParams() throws IllegalArgumentExcept
         int lastLegIndex = itinerary.legs.size() - 1;
         from = itinerary.legs.get(0).from;
         to = itinerary.legs.get(lastLegIndex).to;
+        Map<String, String> parsedQueryParams = this.parseQueryParams();
 
-        // Update actual query modes from the itinerary.
-        updateModeInQueryParams();
+        // Update modes in query params with set derived from itinerary. This ensures that a potentially inaccurate value
+        // in query params is not used when monitoring trips in the future.
+        Set<String> modes = ItineraryUtils.deriveModesFromItinerary(itinerary);
+        parsedQueryParams.put(MODE_PARAM, String.join(",", modes));
+        this.queryParams = ItineraryUtils.toQueryString(parsedQueryParams);
```

Comment:
I would keep these inside `updateModeInQueryParams` so it read like an action (like `clearRealtimeInfo`).